README.md

```diff
@@ -167,23 +167,6 @@ For pure Python projects, OpenAssetIO is available on PyPI, simply run:
 python -m pip install openassetio
 ```
 
-### Other build methods
```

Comment:
[fix] This wasn't really duplicated, it was providing links to `BUILDING.md`, which this change removes.  We can simplify this section to mainly be a link to `BUILDING.md`.

Reply:
I think this section being included again was a rebase error, from my recollection, in reference to [this conversation](https://github.com/OpenAssetIO/OpenAssetIO/pull/711#discussion_r1013063173)

In any case, the bulk of it, the note especially, is just a rewording of the note above, which provides a link to building. The only additional information the deleted segment provides is a gesture out to the containerised builds, which are linked to from BUILDING, and I don't think need to be mentioned in the top level readme.

Perhaps just a sentence at the bottom of `Getting OpenAssetIO`
> Full instructions on how to get OpenAssetIO via a from-source build may be found [here](Building.md)

?

Though even that seems a tad redundant.

Reply:
What do you think about [0b20199](https://github.com/OpenAssetIO/OpenAssetIO/pull/735/commits/0b201994be79121e68a8144e83d560ef1d3354d7)
?

Reply:
Apologies, my bad, I didn't see the `Note` above - for some reason my eyes skip over them.  This is fine (can drop this commit if you prefer, but I think it's useful for people who skip over `Note`s like me :grin: )